build-farm/platform-specific-configurations/mac.sh

```diff
@@ -22,10 +22,18 @@ source "$SCRIPT_DIR/../../sbin/common/constants.sh"
 export MACOSX_DEPLOYMENT_TARGET=10.9
 export BUILD_ARGS="${BUILD_ARGS}"
 
+## JDK8 only: If, at this point in the build, the architecure of the machine is arm64 while the ARCHITECTURE variable
+## is x64 then we need to add the cross compilation option --openjdk-target=x86_64-apple-darwin
+MACHINEARCHITECURE=$(uname -a)
```

Comment:
Missing the second `T` in `ARCHITECTURE` in the variable name
```suggestion
MACHINEARCHITECTURE=$(uname -a)
```

Reply:
I'm slightly surprised we don't already have a variable for this one.

Reply:
The existing `ARCHITECTURE` variable also checks the architecture of the machine [here](https://github.com/adoptium/temurin-build/blob/751e5c8490e556e6c2db55018b048eb0e2f04eea/build-farm/make-adopt-build-farm.sh#L26) but only if it is empty by the time the `make-adopt-build-farm.sh` runs. I believe its value is passed to it by [jenkins](https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-mac-x64-temurin/build?delay=0sec) when you kick off the build
```
{
    "ARCHITECTURE": "x64",
    "TARGET_OS": "mac",
    "VARIANT": "temurin",
    "JAVA_TO_BUILD": "jdk8u",
```
Technically this `ARCHITECTURE` captures the architecture of the binary to build, not of the machine. So if we want to run an x64 build pipeline on an arm64 machine there needs to be a separate `MACHINEARCHITECTURE` variable